queries/c_sharp/indents.scm

```diff
@@ -0,0 +1,117 @@
+[
+ (property_declaration)
+ ; (field_declaration)
+ (variable_declaration)
+ (accessor_declaration)
+
+ ; (declaration_list)
+
+ (parameter_list)
+ (type_parameter_list)
+ (accessor_list)
+ (attribute_argument_list)
+ (base_list)
+ (argument_list)
+
+ (binary_expression)
+ (expression_statement)
+ (return_statement)
+ (local_declaration_statement)
+ (local_function_statement)
+ (yield_statement)
+ (fixed_statement)
+ (unsafe_statement)
+ (labeled_statement)
+ (for_each_statement)
+ (for_statement)
+ (if_statement)
+ (switch_expression)
+ (switch_section)
+ (switch_statement)
+ (while_statement)
+ (lambda_expression)
+ (do_statement)
+ (checked_statement)
+ (try_statement)
+ (using_statement)
+ (local_function_statement)
+ (object_creation_expression)
+ (array_creation_expression)
+ (implicit_object_creation_expression)
+
+ ] @indent.begin
+
+(method_declaration body: _ @indent.begin)
+(namespace_declaration body: _ @indent.begin)
+(record_declaration body: _ @indent.begin)
+(class_declaration body: _ @indent.begin)
+(struct_declaration body: _ @indent.begin)
+(constructor_declaration body: _ @indent.begin)
+(interface_declaration body: _ @indent.begin)
+(enum_declaration body: _ @indent.begin)
+
+(property_declaration (accessor_list "{" @indent.branch) @indent.dedent)
+
+(switch_section
+  (case_switch_label) (case_switch_label) @indent.branch)
```

Comment:
Is there any reason why only switch sections with 2 cases have branches, and only the second one is auto-dedented

Reply:
Because:
```csharp
switch (var) {
    case "a": foo(); break;
}
```
is one `switch_section` starting on `case:` line. So `(switch_section) @indent.begin` start indenting on next line.

 ```csharp
switch (var) {
    case "b":
    case "a": foo(); break;
}
```
is also one `switch_section` with two `case_switch_label`s so if i don't deden't the next ones I end up with second case line indented too much.

Reply:
Couldn't it be something like

```
(switch_body) @indent.begin
"case" @indent.branch
```

Reply:
wouldn't it make it like this?
```csharp
switch (var) {
case "b":
case "a": foo(); break;
}
```
Thus without indenting the cases at all?

Reply:
it's our current version for C.

If that style is not a standard for C# then you can ignore this.

Reply:
Yours is fine :D. I don't think this case_switch is an issue. I have done the same for swift, C, and ocaml and haven't received complaints (yet)